src/main/java/com/google/devtools/build/docgen/templates/be/make-variables.vm

```diff
@@ -17,105 +17,121 @@ title: Make Variables
+<p>
+<code>my_attr = "prefix $(FOO) suffix"</code>
 </p>
 
-<h2 id='make-var-substitution'>"Make" variable substitution</h2>
+<p>
+  In other words, any substring matching <code>$(FOO)</code> gets expanded
+  to <code>FOO</code>'s value. If that value is <code>"bar"</code>, the final
+  string becomes:
+</p>
 
-<p>Variables can be referenced in attributes
+<p>
+<code>my_attr = "prefix bar suffix"</code>
+</p>
 
-  using <code>$(FOO)</code>
-  where <code>FOO</code> is
-  the variable name. In the attribute documentation of rules, it is mentioned
-  when an attribute is subject to "Make" variable substitution. For those
-  attributes this means that any substrings of the form <code>$(X)</code>
-  within those attributes will be interpreted as references to the "Make"
-  variable <var>X</var>, and will be replaced by the appropriate value of that
-  variable for the applicable build configuration. The parens may be omitted
-  for variables whose name is a single character.
+<p>
+  If <code>FOO</code> doesn't correspond to a variable known to the consuming
+  rule, Bazel fails with an error.
 </p>
+
 <p>
-  It is an error if such attributes contain embedded strings of the
-  form <code>$(X)</code> where <var>X</var> is not the name of a
-  "Make" variable, or unclosed references such as <code>$(</code> not
-  matched by a corresponding <code>)</code>.
+  To write <code>$</code> as a a string literal (i.e. to prevent variable
+  expansion), write <code>$$</code>.
 </p>
+
+<h2 id="predefined_variables">Predefined variables</h2>
+
 <p>
-  Within such attributes, literal dollar signs must be escaped
-  as <code>$$</code> to prevent this expansion.
+  Predefined "Make" variables can be referenced by any attribute marked as
+  <i>"Subject to 'Make variable' substitution"</i> on any rule.
 </p>
+
 <p>
-  Those attributes that are subject to this substitution are
-  explicitly indicated as such in their definitions in this document.
+  To see the list of these variables and their values for a given set of build
+  options, run
 </p>
 
-<h2 id="predefined_variables">Predefined "Make" Variables</h2>
+<p><code>bazel info --show_make_env [build options]</code></p>
 
-<p>Bazel defines a set of "Make" variables for you.</p>
+<p>
+  and look at the top output lines with capital letters.
+</p>
 
-<p>The build system also provides a consistent PATH environment variable for
-genrules and tests which need to execute shell commands. For genrules, you can
-indirect your commands using the "Make" variables below.  For basic Unix
-utilities, prefer relying on the PATH environment variable to guarantee correct
-results. For genrules involving compiler and platform invocation, you must use
-the "Make" variable syntax.  The same basic command set is also available during
-tests. Simply rely on the PATH.</p>
+<p><strong>C++ tool path variables</strong></p>
+<p>
+  The following will eventually be converted to <a
```

Comment:
This has happened already. To use C++ make variables, you need to add the following to a target:
toolchains = ["@bazel_tools//tools/cpp:current_cc_toolchain"]

Reply:
Really? I'm able to use `$(CC)` in a `java_binary` without any dependency:

```
$ cat testapp/BUILD
java_binary(
    name = "jl",
    srcs = ["jl.java"],
    javacopts = ["-Dbar=$(CC)"],
    main_class = "foo.bar.Jl",
)

$ bazel build --nobuild //testapp:jl
INFO: Analysed target //testapp:jl (1 packages loaded, 2 targets configured).
INFO: Found 1 target...
INFO: Elapsed time: 0.117s
INFO: 0 processes.
INFO: Build completed successfully, 0 total actions
```

This doesn't work for `$(JAVA)`, which requires me to add the appropriate Java toolchain.

`genrule` is similar. Are these rules specially opted in somehow?

Reply:
FYI the above is with Bazel 0.19.0.

Reply:
Is `--incompatible_disable_cc_configuration_make_variables` relevant?

Reply:
See #6489.

Reply:
Yes, the various Java rules also require CppConfiguration, which provides these. (See https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/rules/java/BazelJavaBinaryRule.java;l=44?q=JavaBinaryRule). This is actually not ideal but it outside the scope of the fix.

You are correct that `--incompatible_disable_cc_configuration_make_variables` is the flag that disabled access to the cpp make variables from the global configuration.

Reply:
Do you mean https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/rules/java/BazelJavaBinaryRule.java;l=91?q=JavaBinaryRule ? I didn't think `requiresConfigurationFragments` by itself causes any new dependencies, especially in the pre-trimming world where all rules include `CppConfiguration` unconditionally.

I can't find any rule where this behavior doesn't apply (tried `java_binary`, `genrule`, and `py_binary`). But they all have implicit dependencies on the C++ toolchain, so I suppose that explains that. Given that I'll make the changes you propose but add a caveat that the dependencies might be built-in.